### PR TITLE
Add module health indicator to popover

### LINKED ISSUE #41: Show module health as a visual indicator in the list
As a Product Owner
I want to know the health of each module
So that I a great understand of my sites technical debt.

## Acceptance criteria

- Module rating is displayed for ~~most recent release~~ default branch only (from addons site)
- Matches designs (circular image indicator and numeric indicator)

## Considerations

- ~~The addons site rating system is currently broken (https://github.com/silverstripe/addons.silverstripe.org/issues/172), and work is nearing completion to replace it with a new system (https://github.com/silverstripe/addons.silverstripe.org/issues/178)~~ done
- ~~The addons site does not have an API to expose the data for this yet so we will need to add one (https://github.com/silverstripe/addons.silverstripe.org/issues/180)~~ done

Designs: https://projects.invisionapp.com/share/ASGU6A7RT8U#/screens/291050684

### PR DESCRIPTION
**Pull request by raissanorth**
This PR adds a health indicator to the popover for each module listed. It also ...

- ... writes the rating data to the database.
- ... refactors the loaders. ApiLoader is the base class - holding most of the logic - which SupportedAddonsLoader and the new ModuleHealthLoader extend from.
- Tests were updated were necessary. As most of the logic resides in the ApiLoader, unit tests focus on this class. A few integration tests were added.
- The rating data itself is retrieved from the new Addons API point introduced in https://github.com/silverstripe/addons.silverstripe.org/commit/67fbac68e936926ebf68a11483baa5136248a201
- The module health indicator symbol is rendered as a diff; applied styling follows the BEM methodology
- The module health information links out to the correlated section on addons.org, e.g. to https://addons.silverstripe.org/add-ons/bringyourownideas/silverstripe-maintenance#rating-breakdown

image
*Module health rating above 70%*

image
*Module health rating between 40% and 70%*

image
*Module health rating below 40%*

Resolves #41

### CODE REVIEW

File: src/Tasks/UpdatePackageInfoTask.php [hunk 1]
```diff
@@ -132,6 +161,11 @@ public function run($request)
                 if (is_array($supportedPackages)) {
                     $package['Supported'] = in_array($package['Name'], $supportedPackages);
                 }
+                if (is_array($moduleHealthInfo)) {
+                    if (isset($moduleHealthInfo[$package['Name']])) {
```
Comment (ScopeyNZ) [inline src/Tasks/UpdatePackageInfoTask.php]: These conditions can be merged.

File: src/Util/ApiLoader.php [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        }
+    }
+
+    /**
+     * @return CacheInterface
+     */
+    public function getCache()
+    {
+        if (!$this->cache) {
+            $this->cache = Injector::inst()->get(CacheInterface::class . '.supportedAddons');
```
Comment (ScopeyNZ) [inline src/Util/ApiLoader.php]: We should use a different namespace now this is an abstract class.

File: tests/Util/ApiLoaderTest.php [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+use BringYourOwnIdeas\Maintenance\Util\ApiLoader;
+use GuzzleHttp\Client;
+use GuzzleHttp\Handler\MockHandler;
+use GuzzleHttp\HandlerStack;
+use GuzzleHttp\Psr7\Response;
+use SilverStripe\Dev\SapphireTest;
+use Symfony\Component\Cache\Simple\NullCache;
+
+/**
+ * @group wip
```
Comment (ScopeyNZ) [inline tests/Util/ApiLoaderTest.php]: WIP?

File: src/Util/ApiLoader.php (hunk 2 above)
Comment (robbieaverill) [reply src/Util/ApiLoader.php]: Do you have a suggestion for a good replacement? Perhaps `BringYourOwnIdeas\Maintenance` for this class and + `\ApiLoader` for its subclasses?

Comment (ScopeyNZ) [reply src/Util/ApiLoader.php]: I meant cache namespace. Ha sorry for the confusion.

Comment (robbieaverill) [reply src/Util/ApiLoader.php]: Ah yep, true

Comment (ScopeyNZ) [reply src/Util/ApiLoader.php]: Probably just a generic cache namespace for all loaders would have been fine as each "loader" only uses one cache key but I don't think it really makes a difference.

Comment (robbieaverill) [reply src/Util/ApiLoader.php]: Yep! Updated

File: client/src/components/ModuleDetails/ModuleHealthIndicator.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+      modifier = 'health-indicator__symbol--green';
+    }
+
+    return classnames('health-indicator__symbol', modifier);
+  }
+
+  /**
+   * Helper method to add additional div for module indicator symbol
+   *
+   * @returns {String}|null
```
Comment (robbieaverill) [inline client/src/components/ModuleDetails/ModuleHealthIndicator.js]: `@returns {String|null}`

File: client/src/legacy/entwine/SiteSummary.js [hunk 5]
```diff
@@ -12,6 +12,7 @@ window.jQuery.entwine('ss', ($) => {
           detailsId={this.attr('id')}
           link={this.data('link')}
           linkTitle={this.data('link-title')}
+          rating={this.data('rating')}
```
Comment (robbieaverill) [inline client/src/legacy/entwine/SiteSummary.js]: This binding will need to be updated when #102 is merged

**Pull request merged**